VP-5110: Add API method for bulk products adding to cart

trueboroda opened the pull request:
New method
see http://localhost:2083/docs/
![image](https://user-images.githubusercontent.com/13153256/94541380-3254f580-0250-11eb-97a0-6dca1b200a44.png)

trueboroda committed:
VP-5110: Add API method for bulk products adding to cart

trueboroda committed:
VP-5110: Add [ValidateAntiForgeryToken] attribute to AddItemsToCart method

pushnitsa commented on VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs:
```diff
@@ -98,6 +98,50 @@ public async Task<ActionResult<int>> GetCartItemsCount()
+
+        [HttpPost("items")]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult<AddItemsToCartResult>> AddItemsToCart([FromQuery] string[] productIds)
```
From body
Action should named BulkAddItemsToCart

pushnitsa replied:
Add new controller. Named like ApiCartDemoController.
We should trying keep original storefront code as is if we can, for potential future updates.

asvishnyakov committed:
Merge branch 'dev' into VP-5110

trueboroda committed:
VP-5110: Use common CartValidator to validate bulk products adding to cart

trueboroda committed:
Merge remote-tracking branch 'origin/dev' into VP-5110

trueboroda committed:
VP-5110: Debugging

trueboroda committed:
Merge remote-tracking branch 'origin/VP-5110' into VP-5110

pushnitsa commented on VirtoCommerce.Storefront.Model/Cart/AddItemsToCartResult.cs:
```diff
@@ -0,0 +1,17 @@
+        public AddItemsToCartResult()
+        {
+            ErrorCodes = new string[] { };
+        }
```
Add empty string after constructor

pushnitsa commented on VirtoCommerce.Storefront.Model/Cart/AddItemsToCartResult.cs:
```diff
@@ -0,0 +1,17 @@
+    {
+        public AddItemsToCartResult()
+        {
+            ErrorCodes = new string[] { };
```
Array.Empty<string>(); instead new string[]

pushnitsa commented on VirtoCommerce.Storefront.Model/Cart/ShoppingCart.cs:
```diff
@@ -53,7 +53,7 @@ public ShoppingCart(Currency currency, Language language)
         /// <see cref="CartValidator"/>
         /// </summary>        
-        public string ValidationRuleSet { get; set; } = "default,strict"; //Achtung!!! It is very important not to use spaces for fluent validation  rule set string, all values after the space will be ignored!
+        public string ValidationRuleSet { get; set; } = "default,strict,shipment"; //Achtung!!! It is very important not to use spaces for fluent validation  rule set string, all values after the space will be ignored!
```
Is it not necessary to change original storefront code. Revert and might be push PR to platform team.

pushnitsa commented on VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs:
```diff
@@ -23,23 +23,26 @@ namespace VirtoCommerce.Storefront.Controllers.Api
     [StorefrontApiRoute("cart")]
     [ResponseCache(CacheProfileName = "None")]
-    public class ApiCartController : StorefrontControllerBase
+    public partial class ApiCartController : StorefrontControllerBase
```
Don't use partial classes, Just create new one.

pushnitsa commented on VirtoCommerce.Storefront/Controllers/Api/ApiCartController.cs:
```diff
@@ -23,23 +23,26 @@ namespace VirtoCommerce.Storefront.Controllers.Api
                                  IOrderModule orderApi, IStorefrontUrlBuilder urlBuilder,
-                                 IEventPublisher publisher, ISubscriptionService subscriptionService)
+                                 IEventPublisher publisher, ISubscriptionService subscriptionService,
+                                 ICartService cartService)
```
Revert all changes

trueboroda committed:
VP-5110: Take into account PR notes

Add empty string after constructor
Array.Empty(); instead new string[]
Is it not necessary to change original storefront code. Revert and might be push PR to platform team.
Don't use partial classes, Just create new one.
Revert all changes in original storefront code
